src/main/java/alex/storage/Storage.java

```diff
@@ -0,0 +1,109 @@
+package alex.storage;
+
+import alex.task.Deadline;
+import alex.task.Event;
+import alex.task.Task;
+import alex.task.Todo;
+
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.File;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+
+/**
+ * Represents the storage where the list of tasks are stored. A Storage object corresponds to
+ * an inventory represented by strings referring to the directory and the data file.
+ */
+public class Storage {
+    public ArrayList<Task> tasks;
+
+    public Storage(String directory, String file) {
+        createDirectory(directory);
+        loadTasksFromFile(file);
+    }
+    /**
+     * Creates a directory if the directory doesn't exist.
+     *
+     * @param directory the directory to create
+     */
+    private void createDirectory(String directory) {
+        File folder = new File(directory);
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+    }
+    /**
+     * Returns the list of tasks that are in the specified file.
+     * Saves the current list of tasks into the specified file.
+     * Creates a new file if the file doesn't exist.
+     *
+     * @param file data file.
+     * @return list of tasks.
+     */
+    public ArrayList<Task> loadTasksFromFile(String file) {
+        ArrayList<Task> list = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String desc;
+            while ((desc = reader.readLine()) != null) {
+
+                Task task;
+                if (desc.startsWith("[T]")) {
+                    String details = desc.substring(6).trim();
+                    task = new Todo(details);
+                } else if (desc.startsWith("[D]")) {
+                    String details = desc.substring(6);
+                    String[] info = details.split("//");
+                    String item = info[0].trim();
+                    String dueInter = info[1].substring(4).trim();
+                    LocalDate dueDate = LocalDate.parse(dueInter);
+                    task = new Deadline(item, dueDate);
+                } else {
+                    String details = desc.substring(6);
+                    String[] info = details.split("//");
+                    String item = info[0].trim();
```

Comment:
You could have split the string with whitespace included like: details.split(" // ") to avoid trimming the string later